TagBot is unable to release multiple versions in one job run

TagBot seems to be unable to create GitHub releases when there are multiple releases that need to be made.

Noticed in [SciMLSensitivity](https://github.com/SciML/SciMLSensitivity.jl).
SciMLSensitivity's TagBot setup was broken for a while, due to which a few releases were not automatically created, and the subsequent tags were not generated.

Running the TagBot workflow, with a `lookback` from (approximately) the last time a GitHub release was made[^1], throws an error while trying to create the GitHub release. The corresponding git tag, however, gets created.
<details><summary>Error stacktrace</summary>

```python
Processing version v7.49.0 (aa07de7ffad1000011b83204f7054b2eb6113a8c)
Generating changelog for version v7.49.0 (aa07de7ffad1000011b83204f7054b2eb6113a8c)
Creating release v7.49.0 at aa07de7ffad1000011b83204f7054b2eb6113a8c
{'server': 'GitHub.com', 'date': 'Thu, 18 Jan 2024 10:59:59 GMT', 'content-type': 'application/json; charset=utf-8', 'transfer-encoding': 'chunked', 'x-github-media-type': 'github.v3; format=json', 'x-accepted-github-permissions': 'contents=write; contents=write,workflows=write', 'x-github-api-version-selected': '2022-11-28', 'x-ratelimit-limit': '1000', 'x-ratelimit-remaining': '196', 'x-ratelimit-reset': '1705576468', 'x-ratelimit-used': '804', 'x-ratelimit-resource': 'core', 'access-control-expose-headers': 'ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset', 'access-control-allow-origin': '*', 'strict-transport-security': 'max-age=31536000; includeSubdomains; preload', 'x-frame-options': 'deny', 'x-content-type-options': 'nosniff', 'x-xss-protection': '0', 'referrer-policy': 'origin-when-cross-origin, strict-origin-when-cross-origin', 'content-security-policy': "default-src 'none'", 'vary': 'Accept-Encoding, Accept, X-Requested-With', 'content-encoding': 'gzip', 'x-github-request-id': '5000:137E:C1BAE3:19273D6:65A904AF'}
Error: GitHub returned a 403 permissions-related error.
                    Please check that your ssh key and TagBot permissions are up to date
                    https://github.com/JuliaRegistries/TagBot#setup
                    
Traceback (most recent call last):
  File "/root/tagbot/action/__main__.py", line 96, in <module>
    repo.create_release(version, sha)
  File "/root/tagbot/action/repo.py", line 607, in create_release
    self._repo.create_git_release(
  File "/usr/local/lib/python3.8/site-packages/github/Repository.py", line 1112, in create_git_release
    headers, data = self._requester.requestJsonAndCheck(
  File "/usr/local/lib/python3.8/site-packages/github/Requester.py", line 398, in requestJsonAndCheck
    return self.__check(
  File "/usr/local/lib/python3.8/site-packages/github/Requester.py", line 423, in __check
    raise self.__createException(status, responseHeaders, output)
github.GithubException.GithubException: 403 {"message": "Resource not accessible by integration", "documentation_url": "https://docs.github.com/rest/releases/releases#create-a-release"}

Traceback (most recent call last):
  File "/root/tagbot/action/__main__.py", line 96, in <module>
    repo.create_release(version, sha)
  File "/root/tagbot/action/repo.py", line 607, in create_release
    self._repo.create_git_release(
  File "/usr/local/lib/python3.8/site-packages/github/Repository.py", line 1112, in create_git_release
    headers, data = self._requester.requestJsonAndCheck(
  File "/usr/local/lib/python3.8/site-packages/github/Requester.py", line 398, in requestJsonAndCheck
    return self.__check(
  File "/usr/local/lib/python3.8/site-packages/github/Requester.py", line 423, in __check
    raise self.__createException(status, responseHeaders, output)
github.GithubException.GithubException: 403 {"message": "Resource not accessible by integration", "documentation_url": "https://docs.github.com/rest/releases/releases#create-a-release"}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/root/tagbot/action/__main__.py", line 99, in <module>
    repo.handle_error(e)
  File "/root/tagbot/action/repo.py", line 645, in handle_error
    raise Abort("Cannot continue due to internal failure")
tagbot.action.Abort: Cannot continue due to internal failure
Error: Issue reporting failed
Traceback (most recent call last):
  File "/root/tagbot/action/__main__.py", line 96, in <module>
    repo.create_release(version, sha)
  File "/root/tagbot/action/repo.py", line 607, in create_release
    self._repo.create_git_release(
  File "/usr/local/lib/python3.8/site-packages/github/Repository.py", line 1112, in create_git_release
    headers, data = self._requester.requestJsonAndCheck(
  File "/usr/local/lib/python3.8/site-packages/github/Requester.py", line 398, in requestJsonAndCheck
    return self.__check(
  File "/usr/local/lib/python3.8/site-packages/github/Requester.py", line 423, in __check
    raise self.__createException(status, responseHeaders, output)
github.GithubException.GithubException: 403 {"message": "Resource not accessible by integration", "documentation_url": "https://docs.github.com/rest/releases/releases#create-a-release"}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/docker/api/client.py", line 214, in _retrieve_server_version
    return self.version(api_version=False)["ApiVersion"]
  File "/usr/local/lib/python3.8/site-packages/docker/api/daemon.py", line 181, in version
    return self._result(self._get(url), json=True)
  File "/usr/local/lib/python3.8/site-packages/docker/utils/decorators.py", line 46, in inner
    return f(self, *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/docker/api/client.py", line 237, in _get
    return self.get(url, **self._set_request_timeout(kwargs))
  File "/usr/local/lib/python3.8/site-packages/requests/sessions.py", line 602, in get
    return self.request("GET", url, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.8/site-packages/requests/sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/requests/adapters.py", line 486, in send
    resp = conn.urlopen(
  File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 791, in urlopen
    response = self._make_request(
  File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 497, in _make_request
    conn.request(
TypeError: request() got an unexpected keyword argument 'chunked'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/root/tagbot/action/repo.py", line 640, in handle_error
    self._report_error(trace)
  File "/root/tagbot/action/repo.py", line 442, in _report_error
    "image": self._image_id(),
  File "/root/tagbot/action/repo.py", line [43](https://github.com/SciML/SciMLSensitivity.jl/actions/runs/7568969251/job/20611271359#step:3:44)1, in _image_id
    client = docker.from_env()
  File "/usr/local/lib/python3.8/site-packages/docker/client.py", line 96, in from_env
    return cls(
  File "/usr/local/lib/python3.8/site-packages/docker/client.py", line [45](https://github.com/SciML/SciMLSensitivity.jl/actions/runs/7568969251/job/20611271359#step:3:46), in __init__
    self.api = APIClient(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/docker/api/client.py", line 1[97](https://github.com/SciML/SciMLSensitivity.jl/actions/runs/7568969251/job/20611271359#step:3:98), in __init__
    self._version = self._retrieve_server_version()
  File "/usr/local/lib/python3.8/site-packages/docker/api/client.py", line 221, in _retrieve_server_version
    raise DockerException(
docker.errors.DockerException: Error while fetching server API version: request() got an unexpected keyword argument 'chunked'

0s
```
</details>

However, triggering the same workflow, with the `lookback` adjusted to include only the latest release `v7.52.0`, runs successfully[^2], with the tag and [release being created](https://github.com/SciML/SciMLSensitivity.jl/releases/tag/v7.52.0) as expected.

[^1]: https://github.com/SciML/SciMLSensitivity.jl/actions/runs/7568969251/job/20611271359#step:3:47
[^2]: https://github.com/SciML/SciMLSensitivity.jl/actions/runs/7568953034/job/20611326603#step:3:43

Similar/same message here. This is for a monorepo (SnoopCompile):

<details>
<summary> TagBot.yml log file </summary>

2025-08-28T22:07:58.1801996Z Current runner version: '2.328.0'
2025-08-28T22:07:58.1827091Z ##[group]Runner Image Provisioner
2025-08-28T22:07:58.1828123Z Hosted Compute Agent
2025-08-28T22:07:58.1828754Z Version: 20250825.382
2025-08-28T22:07:58.1829345Z Commit: 7109f2abf901479dac39397456e363ac0cca0730
2025-08-28T22:07:58.1829991Z Build Date: 2025-08-25T22:55:50Z
2025-08-28T22:07:58.1830654Z ##[endgroup]
2025-08-28T22:07:58.1831218Z ##[group]Operating System
2025-08-28T22:07:58.1831740Z Ubuntu
2025-08-28T22:07:58.1832432Z 24.04.3
2025-08-28T22:07:58.1832883Z LTS
2025-08-28T22:07:58.1833302Z ##[endgroup]
2025-08-28T22:07:58.1833852Z ##[group]Runner Image
2025-08-28T22:07:58.1834420Z Image: ubuntu-24.04
2025-08-28T22:07:58.1834889Z Version: 20250824.1.0
2025-08-28T22:07:58.1835953Z Included Software: https://github.com/actions/runner-images/blob/ubuntu24/20250824.1/images/ubuntu/Ubuntu2404-Readme.md
2025-08-28T22:07:58.1837480Z Image Release: https://github.com/actions/runner-images/releases/tag/ubuntu24%2F20250824.1
2025-08-28T22:07:58.1838740Z ##[endgroup]
2025-08-28T22:07:58.1841181Z ##[group]GITHUB_TOKEN Permissions
2025-08-28T22:07:58.1843859Z Actions: write
2025-08-28T22:07:58.1844413Z Attestations: write
2025-08-28T22:07:58.1844886Z Checks: write
2025-08-28T22:07:58.1845450Z Contents: write
2025-08-28T22:07:58.1845925Z Deployments: write
2025-08-28T22:07:58.1846477Z Discussions: write
2025-08-28T22:07:58.1847037Z Issues: write
2025-08-28T22:07:58.1847484Z Metadata: read
2025-08-28T22:07:58.1848422Z Models: read
2025-08-28T22:07:58.1849015Z Packages: write
2025-08-28T22:07:58.1849488Z Pages: write
2025-08-28T22:07:58.1850013Z PullRequests: write
2025-08-28T22:07:58.1850551Z RepositoryProjects: write
2025-08-28T22:07:58.1851135Z SecurityEvents: write
2025-08-28T22:07:58.1851765Z Statuses: write
2025-08-28T22:07:58.1852280Z ##[endgroup]
2025-08-28T22:07:58.1854824Z Secret source: Actions
2025-08-28T22:07:58.1855896Z Prepare workflow directory
2025-08-28T22:07:58.2184898Z Prepare all required actions
2025-08-28T22:07:58.2222682Z Getting action download info
2025-08-28T22:07:58.6880292Z Download action repository 'JuliaRegistries/TagBot@v1' (SHA:62dd7cd21dcde087eca10e70c3fc3f8cefbe44a2)
2025-08-28T22:07:59.2508268Z Complete job name: TagBot
2025-08-28T22:07:59.3105798Z ##[group]Pull down action image 'ghcr.io/juliaregistries/tagbot:1.20.2'
2025-08-28T22:07:59.3158400Z ##[command]/usr/bin/docker pull ghcr.io/juliaregistries/tagbot:1.20.2
2025-08-28T22:08:00.0163008Z 1.20.2: Pulling from juliaregistries/tagbot
2025-08-28T22:08:00.0163946Z 61ffccc6e275: Pulling fs layer
2025-08-28T22:08:00.0164513Z f778d6c4f6e9: Pulling fs layer
2025-08-28T22:08:00.0165090Z d2f62848b12d: Pulling fs layer
2025-08-28T22:08:00.0165632Z 997762e48473: Pulling fs layer
2025-08-28T22:08:00.0166184Z 40f8fd235556: Pulling fs layer
2025-08-28T22:08:00.0166733Z 7300472ba451: Pulling fs layer
2025-08-28T22:08:00.0167317Z c332056b3f65: Pulling fs layer
2025-08-28T22:08:00.0168251Z 1050a7a38fb0: Pulling fs layer
2025-08-28T22:08:00.0169286Z 6abe5de24857: Pulling fs layer
2025-08-28T22:08:00.0170172Z 997762e48473: Waiting
2025-08-28T22:08:00.0170694Z c332056b3f65: Waiting
2025-08-28T22:08:00.0171188Z 40f8fd235556: Waiting
2025-08-28T22:08:00.0171653Z 7300472ba451: Waiting
2025-08-28T22:08:00.0172243Z 1050a7a38fb0: Waiting
2025-08-28T22:08:00.0172811Z 6abe5de24857: Waiting
2025-08-28T22:08:00.2152822Z f778d6c4f6e9: Download complete
2025-08-28T22:08:00.2430311Z d2f62848b12d: Verifying Checksum
2025-08-28T22:08:00.2434072Z d2f62848b12d: Download complete
2025-08-28T22:08:00.2973863Z 61ffccc6e275: Verifying Checksum
2025-08-28T22:08:00.2975546Z 61ffccc6e275: Download complete
2025-08-28T22:08:00.3748292Z 997762e48473: Verifying Checksum
2025-08-28T22:08:00.3749524Z 997762e48473: Download complete
2025-08-28T22:08:00.4759072Z 7300472ba451: Verifying Checksum
2025-08-28T22:08:00.4761204Z 7300472ba451: Download complete
2025-08-28T22:08:00.6049274Z c332056b3f65: Verifying Checksum
2025-08-28T22:08:00.6050655Z c332056b3f65: Download complete
2025-08-28T22:08:00.6247666Z 1050a7a38fb0: Verifying Checksum
2025-08-28T22:08:00.6249542Z 1050a7a38fb0: Download complete
2025-08-28T22:08:00.7678207Z 6abe5de24857: Verifying Checksum
2025-08-28T22:08:00.7681083Z 6abe5de24857: Download complete
2025-08-28T22:08:01.1627691Z 40f8fd235556: Verifying Checksum
2025-08-28T22:08:01.1629215Z 40f8fd235556: Download complete
2025-08-28T22:08:01.4489261Z 61ffccc6e275: Pull complete
2025-08-28T22:08:01.7572160Z f778d6c4f6e9: Pull complete
2025-08-28T22:08:02.3637773Z d2f62848b12d: Pull complete
2025-08-28T22:08:02.3771591Z 997762e48473: Pull complete
2025-08-28T22:08:03.6270402Z 40f8fd235556: Pull complete
2025-08-28T22:08:03.6394095Z 7300472ba451: Pull complete
2025-08-28T22:08:04.7258670Z c332056b3f65: Pull complete
2025-08-28T22:08:04.7381970Z 1050a7a38fb0: Pull complete
2025-08-28T22:08:04.7557449Z 6abe5de24857: Pull complete
2025-08-28T22:08:04.7591308Z Digest: sha256:54fc0415529ea8e2ba5634b6e478fd33bdb4259d9d74fcceb58210e6bba49b88
2025-08-28T22:08:04.7609229Z Status: Downloaded newer image for ghcr.io/juliaregistries/tagbot:1.20.2
2025-08-28T22:08:04.7621922Z ghcr.io/juliaregistries/tagbot:1.20.2
2025-08-28T22:08:04.7645409Z ##[endgroup]
2025-08-28T22:08:04.7925650Z ##[group]Run JuliaRegistries/TagBot@v1
2025-08-28T22:08:04.7926291Z with:
2025-08-28T22:08:04.7926804Z   token: ***
2025-08-28T22:08:04.8027048Z   ssh: ***
2025-08-28T22:08:04.8027258Z   registry: JuliaRegistries/General
2025-08-28T22:08:04.8027593Z   github: github.com
2025-08-28T22:08:04.8027804Z   github_api: api.github.com
2025-08-28T22:08:04.8028019Z   branches: false
2025-08-28T22:08:04.8028196Z   dispatch: false
2025-08-28T22:08:04.8028384Z   dispatch_delay: 5
2025-08-28T22:08:04.8028576Z   lookback: 3
2025-08-28T22:08:04.8028855Z   draft: false
2025-08-28T22:08:04.8029097Z   user: github-actions[bot]
2025-08-28T22:08:04.8029397Z   email: 41898282+github-actions[bot]@users.noreply.github.com
2025-08-28T22:08:04.8031377Z   changelog: ## {{ package }} {{ version }}

{% if previous_release %}
[Diff since {{ previous_release }}]({{ compare_url }})
{% endif %}

{% if custom %}
{{ custom }}
{% endif %}

{% if backport %}
This release has been identified as a backport.
Automated changelogs for backports tend to be wildly incorrect.
Therefore, the list of issues and pull requests is hidden.
<!--
{% endif %}
{% if pulls %}
**Merged pull requests:**
{% for pull in pulls %}
- {{ pull.title }} (#{{ pull.number }}) (@{{ pull.author.username }})
{% endfor %}
{% endif %}

{% if issues %}
**Closed issues:**
{% for issue in issues %}
- {{ issue.title }} (#{{ issue.number }})
{% endfor %}
{% endif %}

{% if backport %}
-->
{% endif %}

2025-08-28T22:08:04.8033711Z ##[endgroup]
2025-08-28T22:08:04.8265162Z ##[command]/usr/bin/docker run --name ghcriojuliaregistriestagbot1202_9c3c8c --label 37d0a8 --workdir /github/workspace --rm -e "INPUT_TOKEN" -e "INPUT_SSH" -e "INPUT_REGISTRY" -e "INPUT_GITHUB" -e "INPUT_GITHUB_API" -e "INPUT_BRANCHES" -e "INPUT_DISPATCH" -e "INPUT_DISPATCH_DELAY" -e "INPUT_LOOKBACK" -e "INPUT_SSH_PASSWORD" -e "INPUT_GPG" -e "INPUT_GPG_PASSWORD" -e "INPUT_REGISTRY_SSH" -e "INPUT_DRAFT" -e "INPUT_USER" -e "INPUT_EMAIL" -e "INPUT_BRANCH" -e "INPUT_SUBDIR" -e "INPUT_TAG_PREFIX" -e "INPUT_CHANGELOG" -e "INPUT_CHANGELOG_IGNORE" -e "HOME" -e "GITHUB_JOB" -e "GITHUB_REF" -e "GITHUB_SHA" -e "GITHUB_REPOSITORY" -e "GITHUB_REPOSITORY_OWNER" -e "GITHUB_REPOSITORY_OWNER_ID" -e "GITHUB_RUN_ID" -e "GITHUB_RUN_NUMBER" -e "GITHUB_RETENTION_DAYS" -e "GITHUB_RUN_ATTEMPT" -e "GITHUB_ACTOR_ID" -e "GITHUB_ACTOR" -e "GITHUB_WORKFLOW" -e "GITHUB_HEAD_REF" -e "GITHUB_BASE_REF" -e "GITHUB_EVENT_NAME" -e "GITHUB_SERVER_URL" -e "GITHUB_API_URL" -e "GITHUB_GRAPHQL_URL" -e "GITHUB_REF_NAME" -e "GITHUB_REF_PROTECTED" -e "GITHUB_REF_TYPE" -e "GITHUB_WORKFLOW_REF" -e "GITHUB_WORKFLOW_SHA" -e "GITHUB_REPOSITORY_ID" -e "GITHUB_TRIGGERING_ACTOR" -e "GITHUB_WORKSPACE" -e "GITHUB_ACTION" -e "GITHUB_EVENT_PATH" -e "GITHUB_ACTION_REPOSITORY" -e "GITHUB_ACTION_REF" -e "GITHUB_PATH" -e "GITHUB_ENV" -e "GITHUB_STEP_SUMMARY" -e "GITHUB_STATE" -e "GITHUB_OUTPUT" -e "RUNNER_OS" -e "RUNNER_ARCH" -e "RUNNER_NAME" -e "RUNNER_ENVIRONMENT" -e "RUNNER_TOOL_CACHE" -e "RUNNER_TEMP" -e "RUNNER_WORKSPACE" -e "ACTIONS_RUNTIME_URL" -e "ACTIONS_RUNTIME_TOKEN" -e "ACTIONS_CACHE_URL" -e "ACTIONS_RESULTS_URL" -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/SnoopCompile.jl/SnoopCompile.jl":"/github/workspace" ghcr.io/juliaregistries/tagbot:1.20.2
2025-08-28T22:08:14.7338169Z Processing version v3.1.3 (6730cf655af35a7882f557273ed9a4ea4dfa7dfb)
2025-08-28T22:08:15.0000554Z Generating changelog for version v3.1.3 (6730cf655af35a7882f557273ed9a4ea4dfa7dfb)
2025-08-28T22:08:26.2249048Z Creating release v3.1.3 at 6730cf655af35a7882f557273ed9a4ea4dfa7dfb
2025-08-28T22:08:26.6407820Z Request POST /repos/timholy/SnoopCompile.jl/releases failed with 403: Forbidden
2025-08-28T22:08:26.6459447Z HTTPHeaderDict({'Date': 'Thu, 28 Aug 2025 22:08:26 GMT', 'Content-Type': 'application/json; charset=utf-8', 'X-GitHub-Media-Type': 'github.v3; format=json', 'x-accepted-github-permissions': 'contents=write; contents=write,workflows=write', 'x-github-api-version-selected': '2022-11-28', 'X-RateLimit-Limit': '5000', 'X-RateLimit-Remaining': '4947', 'X-RateLimit-Reset': '1756421976', 'X-RateLimit-Used': '53', 'X-RateLimit-Resource': 'core', 'Access-Control-Expose-Headers': 'ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset', 'Access-Control-Allow-Origin': '*', 'Strict-Transport-Security': 'max-age=31536000; includeSubdomains; preload', 'X-Frame-Options': 'deny', 'X-Content-Type-Options': 'nosniff', 'X-XSS-Protection': '0', 'Referrer-Policy': 'origin-when-cross-origin, strict-origin-when-cross-origin', 'Content-Security-Policy': "default-src 'none'", 'Vary': 'Accept-Encoding, Accept, X-Requested-With', 'Content-Encoding': 'gzip', 'Transfer-Encoding': 'chunked', 'Server': 'github.com', 'X-GitHub-Request-Id': 'D83B:343583:7E47A2:809F48:68B0D35A'})
2025-08-28T22:08:26.6487270Z ##[error]GitHub returned a 403 permissions-related error.
                    Please check that your ssh key and TagBot permissions are up to date
                    https://github.com/JuliaRegistries/TagBot#setup
                    
2025-08-28T22:08:26.6495292Z Traceback (most recent call last):
2025-08-28T22:08:26.6495663Z   File "/root/tagbot/action/__main__.py", line 96, in <module>
2025-08-28T22:08:26.6495987Z     repo.create_release(version, sha)
2025-08-28T22:08:26.6496234Z     ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^
2025-08-28T22:08:26.6496526Z   File "/root/tagbot/action/repo.py", line 608, in create_release
2025-08-28T22:08:26.6496858Z     self._repo.create_git_release(
2025-08-28T22:08:26.6497196Z     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
2025-08-28T22:08:26.6497825Z         version_tag, version_tag, log, target_commitish=target, draft=self._draft
2025-08-28T22:08:26.6498207Z         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-08-28T22:08:26.6498571Z     )
2025-08-28T22:08:26.6498865Z     ^
2025-08-28T22:08:26.6499459Z   File "/usr/local/lib/python3.13/site-packages/github/Repository.py", line 1527, in create_git_release
2025-08-28T22:08:26.6500582Z     headers, data = self._requester.requestJsonAndCheck("POST", f"{self.url}/releases", input=post_parameters)
2025-08-28T22:08:26.6501202Z                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-08-28T22:08:26.6501733Z   File "/usr/local/lib/python3.13/site-packages/github/Requester.py", line 625, in requestJsonAndCheck
2025-08-28T22:08:26.6502358Z     *self.requestJson(
2025-08-28T22:08:26.6502667Z      ~~~~~~~~~~~~~~~~^
2025-08-28T22:08:26.6502869Z         verb,
2025-08-28T22:08:26.6503305Z         ^^^^^
2025-08-28T22:08:26.6503581Z     ...<5 lines>...
2025-08-28T22:08:26.6503791Z         follow_302_redirect=follow_302_redirect,
2025-08-28T22:08:26.6504048Z         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-08-28T22:08:26.6504358Z     )
2025-08-28T22:08:26.6504556Z     ^
2025-08-28T22:08:26.6504913Z   File "/usr/local/lib/python3.13/site-packages/github/Requester.py", line 959, in requestJson
2025-08-28T22:08:26.6505501Z     status, responseHeaders, output = self.__requestEncode(
2025-08-28T22:08:26.6505960Z                                       ~~~~~~~~~~~~~~~~~~~~^
2025-08-28T22:08:26.6506497Z         cnx, verb, url, parameters, headers, input, encode, follow_302_redirect=follow_302_redirect
2025-08-28T22:08:26.6507090Z         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-08-28T22:08:26.6507346Z     )
2025-08-28T22:08:26.6507799Z     ^
2025-08-28T22:08:26.6508391Z   File "/usr/local/lib/python3.13/site-packages/github/Requester.py", line 1098, in __requestEncode
2025-08-28T22:08:26.6509191Z     status, responseHeaders, output = self.__requestRaw(
2025-08-28T22:08:26.6509519Z                                       ~~~~~~~~~~~~~~~~~^
2025-08-28T22:08:26.6510088Z         cnx, verb, url, requestHeaders, encoded_input, stream=stream, follow_302_redirect=follow_302_redirect
2025-08-28T22:08:26.6510539Z         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-08-28T22:08:26.6510877Z     )
2025-08-28T22:08:26.6511021Z     ^
2025-08-28T22:08:26.6511383Z   File "/usr/local/lib/python3.13/site-packages/github/Requester.py", line 1136, in __requestRaw
2025-08-28T22:08:26.6511890Z     response = cnx.getresponse()
2025-08-28T22:08:26.6512376Z   File "/usr/local/lib/python3.13/site-packages/github/Requester.py", line 214, in getresponse
2025-08-28T22:08:26.6512787Z     r = verb(
2025-08-28T22:08:26.6512952Z         url,
2025-08-28T22:08:26.6513200Z     ...<4 lines>...
2025-08-28T22:08:26.6513455Z         allow_redirects=False,
2025-08-28T22:08:26.6513827Z     )
2025-08-28T22:08:26.6514243Z   File "/usr/local/lib/python3.13/site-packages/requests/sessions.py", line 637, in post
2025-08-28T22:08:26.6514912Z     return self.request("POST", url, data=data, json=json, **kwargs)
2025-08-28T22:08:26.6515254Z            ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-08-28T22:08:26.6515809Z   File "/usr/local/lib/python3.13/site-packages/requests/sessions.py", line 589, in request
2025-08-28T22:08:26.6516232Z     resp = self.send(prep, **send_kwargs)
2025-08-28T22:08:26.6516715Z   File "/usr/local/lib/python3.13/site-packages/requests/sessions.py", line 703, in send
2025-08-28T22:08:26.6517368Z     r = adapter.send(request, **kwargs)
2025-08-28T22:08:26.6517991Z   File "/usr/local/lib/python3.13/site-packages/requests/adapters.py", line 667, in send
2025-08-28T22:08:26.6518369Z     resp = conn.urlopen(
2025-08-28T22:08:26.6518683Z         method=request.method,
2025-08-28T22:08:26.6518912Z     ...<9 lines>...
2025-08-28T22:08:26.6519223Z         chunked=chunked,
2025-08-28T22:08:26.6519509Z     )
2025-08-28T22:08:26.6519906Z   File "/usr/local/lib/python3.13/site-packages/urllib3/connectionpool.py", line 934, in urlopen
2025-08-28T22:08:26.6520537Z     retries = retries.increment(method, url, response=response, _pool=self)
2025-08-28T22:08:26.6521197Z   File "/usr/local/lib/python3.13/site-packages/github/GithubRetry.py", line 187, in increment
2025-08-28T22:08:26.6521844Z     raise Requester.createException(response.status, response.headers, content)  # type: ignore
2025-08-28T22:08:26.6522328Z     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-08-28T22:08:26.6523297Z github.GithubException.GithubException: Resource not accessible by integration: 403 {"message": "Resource not accessible by integration", "documentation_url": "https://docs.github.com/rest/releases/releases#create-a-release", "status": "403"}
2025-08-28T22:08:26.6524226Z 
2025-08-28T22:08:30.5782383Z Response (200): {
2025-08-28T22:08:30.5782892Z   "status": "Submitted error report"
2025-08-28T22:08:30.5783408Z }
2025-08-28T22:08:30.5794699Z Traceback (most recent call last):
2025-08-28T22:08:30.5812172Z   File "/root/tagbot/action/__main__.py", line 96, in <module>
2025-08-28T22:08:30.5813174Z     repo.create_release(version, sha)
2025-08-28T22:08:30.5813720Z     ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^
2025-08-28T22:08:30.5814230Z   File "/root/tagbot/action/repo.py", line 608, in create_release
2025-08-28T22:08:30.5814759Z     self._repo.create_git_release(
2025-08-28T22:08:30.5815121Z     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
2025-08-28T22:08:30.5815594Z         version_tag, version_tag, log, target_commitish=target, draft=self._draft
2025-08-28T22:08:30.5816226Z         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-08-28T22:08:30.5816631Z     )
2025-08-28T22:08:30.5816882Z     ^
2025-08-28T22:08:30.5817722Z   File "/usr/local/lib/python3.13/site-packages/github/Repository.py", line 1527, in create_git_release
2025-08-28T22:08:30.5819137Z     headers, data = self._requester.requestJsonAndCheck("POST", f"{self.url}/releases", input=post_parameters)
2025-08-28T22:08:30.5819952Z                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-08-28T22:08:30.5820760Z   File "/usr/local/lib/python3.13/site-packages/github/Requester.py", line 625, in requestJsonAndCheck
2025-08-28T22:08:30.5821443Z     *self.requestJson(
2025-08-28T22:08:30.5821724Z      ~~~~~~~~~~~~~~~~^
2025-08-28T22:08:30.5821977Z         verb,
2025-08-28T22:08:30.5822200Z         ^^^^^
2025-08-28T22:08:30.5822434Z     ...<5 lines>...
2025-08-28T22:08:30.5822725Z         follow_302_redirect=follow_302_redirect,
2025-08-28T22:08:30.5823102Z         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-08-28T22:08:30.5823436Z     )
2025-08-28T22:08:30.5823649Z     ^
2025-08-28T22:08:30.5824160Z   File "/usr/local/lib/python3.13/site-packages/github/Requester.py", line 959, in requestJson
2025-08-28T22:08:30.5825084Z     status, responseHeaders, output = self.__requestEncode(
2025-08-28T22:08:30.5825653Z                                       ~~~~~~~~~~~~~~~~~~~~^
2025-08-28T22:08:30.5826248Z         cnx, verb, url, parameters, headers, input, encode, follow_302_redirect=follow_302_redirect
2025-08-28T22:08:30.5826873Z         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-08-28T22:08:30.5827248Z     )
2025-08-28T22:08:30.5827460Z     ^
2025-08-28T22:08:30.5828175Z   File "/usr/local/lib/python3.13/site-packages/github/Requester.py", line 1098, in __requestEncode
2025-08-28T22:08:30.5828886Z     status, responseHeaders, output = self.__requestRaw(
2025-08-28T22:08:30.5829304Z                                       ~~~~~~~~~~~~~~~~~^
2025-08-28T22:08:30.5829933Z         cnx, verb, url, requestHeaders, encoded_input, stream=stream, follow_302_redirect=follow_302_redirect
2025-08-28T22:08:30.5830618Z         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-08-28T22:08:30.5831011Z     )
2025-08-28T22:08:30.5831230Z     ^
2025-08-28T22:08:30.5831742Z   File "/usr/local/lib/python3.13/site-packages/github/Requester.py", line 1136, in __requestRaw
2025-08-28T22:08:30.5832395Z     response = cnx.getresponse()
2025-08-28T22:08:30.5833080Z   File "/usr/local/lib/python3.13/site-packages/github/Requester.py", line 214, in getresponse
2025-08-28T22:08:30.5833688Z     r = verb(
2025-08-28T22:08:30.5833923Z         url,
2025-08-28T22:08:30.5834146Z     ...<4 lines>...
2025-08-28T22:08:30.5834437Z         allow_redirects=False,
2025-08-28T22:08:30.5834732Z     )
2025-08-28T22:08:30.5835211Z   File "/usr/local/lib/python3.13/site-packages/requests/sessions.py", line 637, in post
2025-08-28T22:08:30.5835999Z     return self.request("POST", url, data=data, json=json, **kwargs)
2025-08-28T22:08:30.5836486Z            ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-08-28T22:08:30.5837324Z   File "/usr/local/lib/python3.13/site-packages/requests/sessions.py", line 589, in request
2025-08-28T22:08:30.5838114Z     resp = self.send(prep, **send_kwargs)
2025-08-28T22:08:30.5838713Z   File "/usr/local/lib/python3.13/site-packages/requests/sessions.py", line 703, in send
2025-08-28T22:08:30.5839309Z     r = adapter.send(request, **kwargs)
2025-08-28T22:08:30.5839905Z   File "/usr/local/lib/python3.13/site-packages/requests/adapters.py", line 667, in send
2025-08-28T22:08:30.5840516Z     resp = conn.urlopen(
2025-08-28T22:08:30.5840901Z         method=request.method,
2025-08-28T22:08:30.5841257Z     ...<9 lines>...
2025-08-28T22:08:30.5841533Z         chunked=chunked,
2025-08-28T22:08:30.5841830Z     )
2025-08-28T22:08:30.5842419Z   File "/usr/local/lib/python3.13/site-packages/urllib3/connectionpool.py", line 934, in urlopen
2025-08-28T22:08:30.5843324Z     retries = retries.increment(method, url, response=response, _pool=self)
2025-08-28T22:08:30.5844200Z   File "/usr/local/lib/python3.13/site-packages/github/GithubRetry.py", line 187, in increment
2025-08-28T22:08:30.5845410Z     raise Requester.createException(response.status, response.headers, content)  # type: ignore
2025-08-28T22:08:30.5846127Z     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-08-28T22:08:30.5847937Z github.GithubException.GithubException: Resource not accessible by integration: 403 {"message": "Resource not accessible by integration", "documentation_url": "https://docs.github.com/rest/releases/releases#create-a-release", "status": "403"}
2025-08-28T22:08:30.5849386Z 
2025-08-28T22:08:30.5849679Z During handling of the above exception, another exception occurred:
2025-08-28T22:08:30.5850101Z 
2025-08-28T22:08:30.5850235Z Traceback (most recent call last):
2025-08-28T22:08:30.5850732Z   File "<frozen runpy>", line 198, in _run_module_as_main
2025-08-28T22:08:30.5851225Z   File "<frozen runpy>", line 88, in _run_code
2025-08-28T22:08:30.5851800Z   File "/root/tagbot/action/__main__.py", line 99, in <module>
2025-08-28T22:08:30.5852301Z     repo.handle_error(e)
2025-08-28T22:08:30.5852618Z     ~~~~~~~~~~~~~~~~~^^^
2025-08-28T22:08:30.5853049Z   File "/root/tagbot/action/repo.py", line 646, in handle_error
2025-08-28T22:08:30.5853625Z     raise Abort("Cannot continue due to internal failure")
2025-08-28T22:08:30.5854216Z tagbot.action.Abort: Cannot continue due to internal failure
2025-08-28T22:08:30.7600435Z Cleaning up orphan processes

</details>

--- a/tagbot/action/__main__.py
+++ b/tagbot/action/__main__.py
@@ -89,11 +89,22 @@ try:
     if gpg:
         repo.configure_gpg(gpg, get_input("gpg_password"))
 
+    errors = []
     for version, sha in versions.items():
-        logger.info(f"Processing version {version} ({sha})")
-        if get_input("branches", "false") == "true":
-            repo.handle_release_branch(version)
-        repo.create_release(version, sha)
+        try:
+            logger.info(f"Processing version {version} ({sha})")
+            if get_input("branches", "false") == "true":
+                repo.handle_release_branch(version)
+            repo.create_release(version, sha)
+        except Exception as e:
+            logger.error(f"Failed to process version {version}: {e}")
+            errors.append((version, e))
+            repo.handle_error(e, raise_abort=False)
+
+    if errors:
+        failed = ", ".join(v for v, _ in errors)
+        logger.error(f"Failed to release versions: {failed}")
+        sys.exit(1)
 except Exception as e:
     try:
         repo.handle_error(e)

--- a/tagbot/action/repo.py
+++ b/tagbot/action/repo.py
@@ -673,7 +673,7 @@ class Repo:
         except Exception as e:
             logger.debug(f"Could not check rate limit: {e}")
 
-    def handle_error(self, e: Exception) -> None:
+    def handle_error(self, e: Exception, *, raise_abort: bool = True) -> None:
         """Handle an unexpected error."""
         allowed = False
         internal = True
@@ -707,7 +707,7 @@ class Repo:
             except Exception:
                 logger.error("Issue reporting failed")
                 logger.info(traceback.format_exc())
-            finally:
+            if raise_abort:
                 raise Abort("Cannot continue due to internal failure")
 
     def commit_sha_of_version(self, version: str) -> Optional[str]:

